### PR TITLE
spool response.Body to a temporary file

### PR DESCRIPTION
**Pull request by gbbr**
I like this tool :-)

Is this anywhere near to what you were aiming for?

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -160,11 +160,16 @@ func visit(url *url.URL) {
...
+	if err != nil {
+		log.Fatalf("failed to create temporary file: %v", err)
+	}
 	t5 := time.Now()
-	if _, err := io.Copy(ioutil.Discard, resp.Body); err != nil {
+	if _, err := io.Copy(f, resp.Body); err != nil {
 		log.Fatalf("failed to read response body: %v", err)
 	}
 	resp.Body.Close()
+	f.Close()
```
Comment (davecheney) [inline main.go]: move this to line 160 and use a defer

File: README.md [hunk 2]
```diff
@@ -27,7 +27,7 @@ The big ticket features which are missing at this point are:
 - ~~support for https; this involves negotating the handshake before sending the request, adjusting the various timing points, and using the alternative template.~~ done, thanks @freeformz
 - ~~detect when `httpstat` is being piped to a file or other not tty device and disable coloured output.~~ done, thanks @amy
 - support for more CURL flags, like `-H HEADER` and `-X METHOD`.
-- spool response.Body to a temporary file.
+- ~~spool response.Body to a temporary file.~~
```
Comment (davecheney) [inline README.md]: dont' forget to add your name

File: main.go [hunk 3]
```diff
@@ -179,7 +184,7 @@ func visit(url *url.URL) {
 		fmt.Println(grayscale(14)(k+":"), color.CyanString(strings.Join(resp.Header[k], ",")))
 	}
 
-	fmt.Println("\nBody discarded\n")
+	fmt.Printf("\n%s stored in: %s\n\n", color.GreenString("Body"), f.Name())
```
Comment (davecheney) [inline main.go]: What about when "-I" is used, the body should not be stored because there is no body.

**Comment by gbbr** [comment]
Right, I've omitted the `-I` flag...

I've amended my commit with some changes to support that. Not sure if this is the optimal solution.

**Comment by gbbr** [comment]
It seems that the body is being read (and discarded) even with the `-I` flag. I'm not sure if that's intended and if I was supposed to change it. Either way, I'll be away from Github for a week, so if this becomes urgent and someone else wants to pick it up, feel free to close ;-)

**Comment by davecheney** [comment]
It's fine to always read from resp.Body because it's guarenteed to be an
io.Reader, not sometimes nil. However we shouldn't create a file on disk to
write 0 bytes in the case where there is no body, like a 30x response, or a
HEAD request.

**Comment by davecheney** [comment]
Hopefully #50 will have made this PR easier to integrate.

**Comment by amy** [comment]
@gbbr Hello! Looks like this has been handled by #48. May want to close your PR.

**Comment by gbbr** [comment]
Great! Thanks!

**Pull request closed without merging**